### PR TITLE
feat: stagenet network

### PR DESCRIPTION
**Pull request by brianp**
Description
---
This PR introduces the genesis block, and consensus for the upcoming StageNet network.

Apart from this PR I've generated seed node identities, which will be added to the seeds after this is merged, and the seeds are brought up.

Motivation and Context
---
Get into release cycle habits and hygiene.

How Has This Been Tested?
---
CI (but I'm expecting some test to fail on first pass due to the default network change).

### CODE REVIEW

File: base_layer/core/src/blocks/genesis_block.rs [hunk 1]
```diff
@@ -63,6 +64,99 @@ pub fn get_genesis_block(network: Network) -> ChainBlock {
     }
 }
 
+pub fn get_stagenet_genesis_block() -> ChainBlock {
```
Comment (sdbondi) [inline base_layer/core/src/blocks/genesis_block.rs:67]: Please add a test for this at the bottom of this file like we do for esme and igor.

File: base_layer/core/src/blocks/genesis_block.rs [hunk 2]
```diff
@@ -63,6 +64,99 @@ pub fn get_genesis_block(network: Network) -> ChainBlock {
...
+    let block = get_stagenet_genesis_block_raw();
+
+    let accumulated_data = BlockHeaderAccumulatedData {
+        hash: block.hash(),
+        total_kernel_offset: block.header.total_kernel_offset.clone(),
+        achieved_difficulty: 1.into(),
```
Comment (SWvheerden) [inline base_layer/core/src/blocks/genesis_block.rs:73]: Is this the actual difficulty of the block?

File: base_layer/core/src/blocks/genesis_block.rs [hunk 3]
```diff
@@ -63,6 +64,99 @@ pub fn get_genesis_block(network: Network) -> ChainBlock {
...
+    let coinbase_meta_sig = CommitmentAndPublicKeySignature::new(
+        Commitment::from_hex("7e236c28aeac44c79a65d855dc678ccc19f960bc0a9c50a2d5bc19ee1df20d23").unwrap(),
+        PublicKey::from_hex("241f5034fd1dda26b223d2b12e816a1f729da86e0faa5609f2258ee6dd2c8e6e").unwrap(),
+        PrivateKey::from_hex("5271e5e4d4c8c4f791a6268aa3b21823a1e56c1c557304853192a876764aa902").unwrap(),
+        PrivateKey::from_hex("75b8fba7ecb555785bca632d142d33d464669c6c6288fb17122a220ca8cceb01").unwrap(),
+        PrivateKey::from_hex("b3000fb10de4c100cc271b3cb5065a916904dc265b37c4c0b45f9cb9a0f7cc00").unwrap(),
+    );
+    let coinbase = TransactionOutput::new(
+        TransactionOutputVersion::get_current_version(),
+        OutputFeatures::create_coinbase(6, None),
```
Comment (SWvheerden) [inline base_layer/core/src/blocks/genesis_block.rs]: Does not match the coinbase lockheight. 
But a test unit test should point this out.

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 4]
```diff
@@ -615,6 +615,53 @@ impl ConsensusConstants {
...
+        algos.insert(PowAlgorithm::Monero, PowAlgorithmConstants {
+            max_target_time: 800,
+            min_difficulty: 70_000_000.into(),
+            max_difficulty: u64::MAX.into(),
+            target_time: 200,
+        });
+        let (input_version_range, output_version_range, kernel_version_range) = version_zero();
+        vec![ConsensusConstants {
+            effective_from_height: 0,
+            coinbase_lock_height: 1,
```
Comment (SWvheerden) [inline base_layer/core/src/consensus/consensus_constants.rs]: We have been using small numbers to make tests easier, but I take it we want to have stagenet be a larger more realistic number?

**Review by SWvheerden — COMMENTED** [review]
Looks good so far

File: base_layer/core/src/consensus/consensus_constants.rs (hunk 4 above)
Comment (brianp) [reply base_layer/core/src/consensus/consensus_constants.rs]: Yeah, I did find my test mining was going a little fast. Who needs to make the call on what these final numbers should be?

Comment (brianp) [reply base_layer/core/src/consensus/consensus_constants.rs]: Moved the numbers to match current esme constatns + larger lock height.

File: base_layer/core/src/blocks/genesis_block.rs (hunk 3 above)
Comment (brianp) [reply base_layer/core/src/blocks/genesis_block.rs]: Updated to new lock height and ensured they match the constants.

File: base_layer/core/src/blocks/genesis_block.rs (hunk 2 above)
Comment (brianp) [reply base_layer/core/src/blocks/genesis_block.rs:73]: It's the difficulty emse has so I left it as is.

File: base_layer/core/src/blocks/genesis_block.rs (hunk 1 above)
Comment (brianp) [reply base_layer/core/src/blocks/genesis_block.rs:67]: Added test, which found errors, that is now fixed. 👍🏻

File: base_layer/core/src/blocks/genesis_block.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/core/src/blocks/genesis_block.rs:73]: Just check if this is the stagenet difficulty as well.

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 5]
```diff
@@ -615,6 +615,58 @@ impl ConsensusConstants {
...
+        let mut algos = HashMap::new();
+        algos.insert(PowAlgorithm::Sha3, PowAlgorithmConstants {
+            max_target_time: 1800,
+            min_difficulty: 60_000_000.into(),
+            max_difficulty: u64::MAX.into(),
+            target_time: 300,
+        });
+        algos.insert(PowAlgorithm::Monero, PowAlgorithmConstants {
+            max_target_time: 1200,
+            min_difficulty: 70_000_000.into(),
```
Comment (stringhandler) [inline base_layer/core/src/consensus/consensus_constants.rs]: I think this is too high and will prevent mining

**Review by stringhandler — COMMENTED** [review]
Happy to merge once the monero difficulty is confirmed

File: base_layer/core/src/consensus/consensus_constants.rs (hunk 5 above)
Comment (brianp) [reply base_layer/core/src/consensus/consensus_constants.rs]: Yeah, looks like this was the placeholder value from `mainnet` and not the value from esme. Updated to match the esme value.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**